waiter/src/waiter/service_description.clj

```diff
@@ -884,10 +884,14 @@
           service-description (compute-effective this service-id core-service-description)
           reference-type->entry (cond-> (or reference-type->entry {})
                                   (seq source-tokens)
-                                  (assoc :token {:sources (map walk/keywordize-keys source-tokens)}))]
+                                  (assoc :token {:sources (map walk/keywordize-keys source-tokens)}))
+          ;; only for exclusive mapping can we be sure that the service will always have the run-as-user coming from the token
+          ;; for legacy mapping services, it is possible for the run-as-user to come from the token or the request
```

Comment:
Thanks for adding this comment. It was helpful for clarifying the condition below.